src/App.tsx

```diff
@@ -1,24 +1,200 @@
+      title,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then((newTodo: Todo) => {
+        setTodos(currentTodos => {
+          return [...currentTodos, newTodo];
+        });
+      })
+      .catch((error) => {
+        setErrorMessage(Errors.UnableToAddATodo);
+        throw error;
+      })
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  }
+
+  function deleteTodo(todoId: number) {
+    setLoadingTodosPause(prev => [...prev, todoId]);
+
+    return todosApiServices.deleteUserTodo(todoId)
+      .then(() => setTodos(currentTodos => currentTodos
+        .filter(todo => todo.id !== todoId)))
+      .catch((error) => {
+        setTodos(todos);
+        setErrorMessage(Errors.UnabletoDeleteATodo);
+        throw error;
+      })
+      .finally(() => {
+        setLoadingTodosPause(
+          prev => prev.filter(loadingId => loadingId !== todoId),
+        );
+      });
+  }
+
+  async function clearCompletedTodos() {
+    const completedTodos
+      = todos?.filter(({ completed }) => completed);
+
+    if (completedTodos?.length) {
+      Promise.all(
+        completedTodos.map((todo) => deleteTodo(todo.id)),
+      )
+        .catch(() => {
+          setErrorMessage(Errors.UnabletoDeleteATodo);
+        });
+    }
+  }
+
+  function updateTodo(updatedTodo: Todo) {
+    setLoadingTodosPause(currentTodo => [...currentTodo, updatedTodo.id]);
+
+    return todosApiServices.updateTodo(updatedTodo)
+      .then(() => setTodos(prev => (
+        prev.map(prevTodo => (
+          prevTodo.id === updatedTodo.id
+            ? updatedTodo
+            : prevTodo
+        ))
+      )))
+      .catch(() => setErrorMessage(Errors.UnableToUpdateATodo))
+      .finally(() => {
+        setLoadingTodosPause(currentTodos => currentTodos
+          .filter(todoId => updatedTodo.id !== todoId));
+      });
+  }
+
+  async function toggleAll() {
+    const isAllCompleted = todos
+      .every(todo => todo.completed);
+
+    const updatedTodos = todos
+      .filter(todo => isAllCompleted === todo.completed);
+
+    await Promise.all(updatedTodos.map(todo => (
+      updateTodo({
+        ...todo,
+        completed: !isAllCompleted,
+      })
+    )));
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this buttons is active only if there are some active todos */}
+          {todos.length > 0 && (
+            <ToggleAll
+              onToggleAll={() => toggleAll()}
```

Comment:
```suggestion
              onToggleAll={toggleAll}
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**

Reply:
Yes, I know about this and this was my first approach. But the crazy settings of ESLint doesn't allow to do this. 
  116:22  error  JSX props should not use functions                   react/jsx-no-bind
  126:15  error  JSX props should not use functions                   react/jsx-no-bind
  147:17  error  JSX props should not use functions

Reply:
@baraban2003 ok, good that u understand how to do that right